src/main/ui/app/route/SearchPage.jsx

```diff
@@ -31,10 +31,16 @@ export default class SearchPage extends Component {
     }
 
     doSearch() {
+        const advancedQuery = this.props.location.query;
         const query = this.props.location.query.q;
         const type = this.props.location.query.type;
-	const tag = this.props.location.query.tag;
-	const searchKey = `type:${type ? type : "all"} query:${query ? query : ""} tag:${tag ? tag : ""}`;
+        const tag = this.props.location.query.tag;
+        const exclude = this.props.location.query.exclude;
+        const moduleid = this.props.location.query.moduleid;
+        const name = this.props.location.query.name;
+        const status = this.props.location.query.status;
```

Comment:
Why do we have to enumerate all of these here? Can't they be encapsulated within SearchBox? E.g. this class only needs to care about a single search string ("query"), right?

Reply:
Sorry you had to see that. It was very foolish of me to do so :)